src/js/load-package.ts

```diff
@@ -64,7 +57,9 @@ async function initializePackageIndex(lockFileURL: string) {
   await loadPackage(API.config.packages, { messageCallback() {} });
 }
 
-API.packageIndexReady = initializePackageIndex(API.config.lockFileURL);
+if (API.lockFilePromise) {
```

Comment:
Is there a case where this is not true though and shouldn't it be an error? or is it just for typing? Maybe worth adding a comment.

Reply:
Yeah well I feel bad about this not being a function call from `pyodide.ts` but it can't be for performance reasons. But I'm also interested in allowing libpyodide to be linked into other applications that possibly don't use the loader logic. Particularly I've been investigating things like linking libpyodide into a Python without dynamic linking, memory snapshotting etc. Mostly what we have is flexible enough that we can just replace `pyodide.ts` and link libpyodide.a but this line causes trouble.